images/win/scripts/ImageHelpers/InstallHelpers.ps1

```diff
@@ -1,124 +1,113 @@
+        $filePath = "msiexec.exe"
+    }
 
+    try
+    {
         Write-Host "Starting Install $Name..."
-        $process = Start-Process -FilePath $FilePath -ArgumentList $ArgumentList -Wait -PassThru
-        $exitCode = $process.ExitCode
+        $process = Start-Process -FilePath $filePath -ArgumentList $ArgumentList -Wait -PassThru
 
+        $exitCode = $process.ExitCode
         if ($exitCode -eq 0 -or $exitCode -eq 3010)
         {
-            Write-Host -Object 'Installation successful'
-            return $exitCode
+            Write-Host "Installation successful"
         }
         else
         {
-            Write-Host -Object "Non zero exit code returned by the installation process : $exitCode."
-            return $exitCode
+            Write-Host "Non zero exit code returned by the installation process: $exitCode"
         }
     }
     catch
     {
-        Write-Host -Object "Failed to install the Executable $Name"
+        Write-Host -Object "Failed to install the $fileExtension $Name"
         Write-Host -Object $_.Exception.Message
-        return -1
+        $exitCode = -1
     }
+
+    return $exitCode
 }
 
 function Stop-SvcWithErrHandling
-<#
-.DESCRIPTION
-Function for stopping the Windows Service with error handling
+{
+    <#
+    .DESCRIPTION
+        Function for stopping the Windows Service with error handling
 
-.AUTHOR
-Andrey Mishechkin v-andmis@microsoft.com
+    .AUTHOR
+        Andrey Mishechkin v-andmis@microsoft.com
 
-.PARAMETER -ServiceName
-The name of stopping service
+    .PARAMETER ServiceName
+        The name of stopping service
 
-.PARAMETER -StopOnError
-Switch for stopping the script and exit from PowerShell if one service is absent
-#>
-{
+    .PARAMETER StopOnError
+        Switch for stopping the script and exit from PowerShell if one service is absent
+    #>
     param (
-        [Parameter(Mandatory, ValueFromPipeLine = $true)] [string] $ServiceName,
-        [Parameter()] [switch] $StopOnError
+        [Parameter(Mandatory, ValueFromPipeLine = $true)]
+        [string] $ServiceName,
+        [switch] $StopOnError
     )
 
-    Process {
-        $Service = Get-Service $ServiceName -ErrorAction SilentlyContinue
-        if (-not $Service) {
+    Process
+    {
+        $service = Get-Service $ServiceName -ErrorAction SilentlyContinue
+        if (-not $service)
+        {
             Write-Warning "[!] Service [$ServiceName] is not found";
-            if ($StopOnError) {
-                exit 1;
+            if ($StopOnError)
+            {
+                exit 1
             }
+
         }
-        else {
+        else
+        {
             Write-Host "Try to stop service [$ServiceName]";
-            try {
+            try
+            {
                 Stop-Service -Name $ServiceName -Force;
-                $Service.WaitForStatus("Stopped", "00:01:00");
+                $service.WaitForStatus("Stopped", "00:01:00");
```

Comment:
Can we get rid of the hardcoded value and use retries approach or something like that?